packages/roosterjs-editor-core/lib/coreApi/selectImage.ts

```diff
@@ -20,19 +28,34 @@ export const selectImage: SelectImage = (core: EditorCore, image: HTMLImageEleme
         addUniqueId(image, IMAGE_ID);
         addUniqueId(core.contentDiv, CONTENT_DIV_ID);
 
+        core.api.selectRange(core, createRange(new Position(image, PositionType.After)));
+
         select(core, image);
-        return {
+
+        const selection: ImageSelectionRange = {
             type: SelectionRangeTypes.ImageSelection,
             ranges: [range],
             image: image,
             areAllCollapsed: range.collapsed,
         };
+
+        core.domEvent.imageSelectionRange = selection;
```

Comment:
also need to update imageSelectionRange when image is null for "unselect" scenario.